e2e/thin_provisioning_test.go

```diff
@@ -0,0 +1,274 @@
+package e2e
+
+import (
+	_ "embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	//	"strconv"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+)
+
+//go:embed testdata/thin_provisioning/pod-template.yaml
+var thinPodTemplateYAML string
+
+//go:embed testdata/thin_provisioning/pvc-template.yaml
+var thinPVCTemplateYAML string
+
+func testThinProvisioning() {
+	testNamespacePrefix := "thinptest-"
+	var ns string
+	BeforeEach(func() {
+		ns = testNamespacePrefix + randomString(10)
+		createNamespace(ns)
+	})
+
+	AfterEach(func() {
+		// When a test fails, I want to investigate the cause. So please don't remove the namespace!
+		if !CurrentGinkgoTestDescription().Failed {
+			kubectl("delete", "namespaces/"+ns)
+		}
+	})
+
+	It("should thin provision a PV", func() {
+		By("deploying Pod with PVC")
+
+		nodeName := "topolvm-e2e-worker"
+		if isDaemonsetLvmdEnvSet() {
+			nodeName = getDaemonsetLvmdNodeName()
+		}
+
+		thinPvcYAML := []byte(fmt.Sprintf(thinPVCTemplateYAML, "thinvol", "1"))
+		stdout, stderr, err := kubectlWithInput(thinPvcYAML, "apply", "-n", ns, "-f", "-")
+		Expect(err).ShouldNot(HaveOccurred(), "stdout=%s, stderr=%s", stdout, stderr)
+
+		thinPodYAML := []byte(fmt.Sprintf(thinPodTemplateYAML, "thinpod", "thinvol", nodeName))
+		stdout, stderr, err = kubectlWithInput(thinPodYAML, "apply", "-n", ns, "-f", "-")
+		Expect(err).ShouldNot(HaveOccurred(), "stdout=%s, stderr=%s", stdout, stderr)
+
+		By("confirming that the lv was created in the thin volume group and pool")
+		var volumeName string
+
+		Eventually(func() error {
+			volumeName, err = getVolumeNameofPVC("thinvol", ns)
+			return err
+		}).Should(Succeed())
+
+		var lv *thinlvinfo
+		Eventually(func() error {
+			fmt.Println("Testing", "getThinLVInfo", "1")
```

Comment:
What does the literal "1" mean here? Is it possible to get rid of it?

Reply:
Sorry. That was for debugging. I will remove it.